atexit.go

```diff
@@ -35,17 +35,28 @@ import (
 
 const (
 	// Version is package version
-	Version = "0.2.0"
+	Version = "0.3.0"
 )
 
-var handlers = []func(){}
+type HandlerId uint
+
+func (h *HandlerId) Get() HandlerId {
+	defer func() {
```

Comment:
This is a nice trick but too clever to my taste. Go is about simplicity and I prefer it here as well. Have a global `var nextID HandlerID` and incement it.

Also, use the lock (or sync/atomic)

Reply:
Feel free to modify the code.
I'll be happy to see how you imagine this.

Reply:
See https://github.com/tebeka/atexit/pull/4